### ISSUE TITLE
Exception: Cannot get fields from a polymorphic serializer given a queryset

### ISSUE THREAD

**Issue by BillBrower** [django-json-api/django-rest-framework-json-api#401]
I'm sure I'm just doing something wrong but the documentation on working with polymorphic resources is pretty sparse so I'm not sure what to change. I have a polymorphic model `Client` with two instance types and when I try to retrieve them using DRF's ModelViewSet I get this error: `Exception: Cannot get fields from a polymorphic serializer given a queryset`.

Here are my views:

```python
from rest_framework import permissions, viewsets
from rest_framework.authentication import TokenAuthentication
from firms.models import Firm, Membership
from .serializers import ClientSerializer

class ClientViewSet(viewsets.ModelViewSet):
    authentication_classes = (TokenAuthentication,)
    included = []
    pagination_class = None
    permission_classes = (permissions.IsAuthenticated,)
    serializer_class = ClientSerializer

    def perform_create(self, serializer):
        membership = Membership.objects.get(user_id=self.request.user.id)
        serializer.save(firm=membership.firm)

    def get_queryset(self):
        try:
            membership = Membership.objects.get(user_id=self.request.user.id)
            return membership.firm.clients.all()
        except Membership.DoesNotExist:
            return []
```

And here is my serializer:

```python
class CorporationSerializer(serializers.ModelSerializer):
    class Meta:
        model = Corporation
    .
    .
    .

class IndividualSerializer(serializers.ModelSerializer):
    class Meta:
        model = Individual
    .
    .
    .

class ClientSerializer(PolymorphicModelSerializer):
    polymorphic_serializers = [CorporationSerializer, IndividualSerializer]

    class Meta:
        model = Client
```

I have my `resource_name`s defined on my models. I've tried passing a list instead of a queryset and that works for an empty response (`[]`) but I run into issue #400 for non-empty responses. What should I be passing to my serializer?

**Comment by mblayman** [comment]
Is this fixed by #402 being merged?

**Comment by BillBrower** [comment]
It does. Thanks for fixing this so quickly!
